spec/unit/outputs/elasticsearch/http_client/pool_spec.rb

```diff
@@ -44,6 +44,25 @@
       expect(subject).to receive(:healthcheck!).once
       sleep(subject.resurrect_delay + 1)
     end
+
+    context "when using an absolute_healthcheck_path option" do
+      let(:options) { super.merge(:absolute_healthcheck_path => true, :healthcheck_path => "http://abc:xyz@localhost:9200")}
+      let(:pool) { described_class.new(logger, adapter, initial_urls, options) }
+
+      before do
+        pool.start
+      end
+
+      after do
+        pool.close
+      end
+
+      it "should use the healthcheck_path as URL to do a health check" do
```

Comment:
look at this feature, I'd probably test that:

1) when `absolute_healthcheck_path` is enabled, `perform_request_to_url` is called with the `url` argument equal to the url constructed from `healthcheck_path`

2) when disabled, it behaves as before: appending the path to the existing hosts url (again observable in `perform_request_to_url`'s arguments

3) what happens when you pass a full URL and disable `absolute_healthcheck_path` (should raise exception)

4) what happens when you pass a path and enable `absolute_healthcheck_path` (should raise exception)